force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +251,18 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
+        Integer yearsDifference;
+        Integer absyearsDifference;
+
+        if (d1 == null || d2 == null) {
+            return null;
+        } else {
+            // Subtract the difference in years
+            yearsDifference = d1.year() - d2.year();
+            absyearsDifference = Math.abs(yearsDifference);
+        }
 
-        // Subtract the difference in years
-
-        return null; // Replace null with the variable you used to store the result
+        return absyearsDifference;
```

Comment:
Good. I often like to leave my main code path of out  of the conditional logic. I think it helps reababilty. 

```
 public static Integer calculateYearDifference(Date d1, Date d2) {
        if (d1 == null || d2 == null) {
            return null;
        }
        return Math.abs(d1.year() - d2.year());
    }
    
```